### PR TITLE
Improve changesets for tree errors

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Improve/add error related changesets for tree.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: .changeset/weak-swans-take.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+
+Improved error reporting
+
+Several cases invalid usage patterns for tree APIs have gained improved error reporting, as well as improved documentation on the APIs detailing what usage is supported.
+These improvements include:
+
+-   Unsupported usages of schema classes: using more than one schema class derived from a single SchemaFactor generated base class. This use to hit internal asserts, but now has a descriptive user facing UsageError.
```
Comment (alexvy86) [inline .changeset/weak-swans-take.md:11]:
```suggestion
Several cases of invalid usage patterns for tree APIs have gained improved error reporting, as well as improved documentation on the APIs detailing what usage is supported.
These improvements include:

-   Unsupported usages of schema classes: using more than one schema class derived from a single SchemaFactory generated base class. This used to hit internal asserts, but now has a descriptive user-facing UsageError.
```

**Review by alexvy86 — COMMENTED** [review]
Just a couple of typos. I know adding new changeset files at this point has some caveats, like they don't end up linked to the right commit. Did we have to do anything about that @tylerbutler ?

**Comment by tylerbutler** [comment]
> Just a couple of typos. I know adding new changeset files at this point has some caveats, like they don't end up linked to the right commit. Did we have to do anything about that @tylerbutler ?

The entries in the per-package changelogs will link to this PR/commit, which isn't ideal but it's the best we can do atm. If there is a better PR to link to, my preferred pattern is to include a sentence at the end of the changeset like, "This change was originally made in PR 1234."

**Comment by CraigMacomber** [comment]
> The entries in the per-package changelogs will link to this PR/commit, which isn't ideal but it's the best we can do atm. If there is a better PR to link to, my preferred pattern is to include a sentence at the end of the changeset like, "This change was originally made in PR 1234."

Done.

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**